### PR TITLE
Mushroom facade

### PR DESCRIPTION
**Pull request by Buvko**

### CODE REVIEW

File: api/src/main/java/cz/muni/fi/pa165/mushrooms/enums/MushroomType.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package cz.muni.fi.pa165.mushrooms.enums;
+
+/**
+ * Type of mushroom.
+ *
+ * @author Buvko
+ */
+public enum MushroomType {
+    EDIBLE, UNEDIBLE, POISONOUS;
+}
```
Comment (bkompis) [inline api/src/main/java/cz/muni/fi/pa165/mushrooms/enums/MushroomType.java]: This duplicates the enum, right? If so, we should remove it from persistence and add 'api' module to dependencies of persistence

File: api/src/main/java/cz/muni/fi/pa165/mushrooms/facade/MushroomFacade.java [hunk 2]
```diff
@@ -1,9 +1,75 @@
...
+import cz.muni.fi.pa165.mushrooms.dto.MushroomDTO;
+import cz.muni.fi.pa165.mushrooms.enums.MushroomType;
+
+import java.util.List;
+
 /**
- * TODO: create  javadoc
  *
- * @author Barbora Kompisova
+ * @author Barbora Kompisova, Buvko
```
Comment (bkompis) [inline api/src/main/java/cz/muni/fi/pa165/mushrooms/facade/MushroomFacade.java:10]: It's all yours, delete my name :)

File: api/src/main/java/cz/muni/fi/pa165/mushrooms/facade/MushroomFacade.java [hunk 3]
```diff
@@ -1,9 +1,75 @@
...
  */
 public interface MushroomFacade {
+
```
Comment (bkompis) [inline api/src/main/java/cz/muni/fi/pa165/mushrooms/facade/MushroomFacade.java:13]: we'll need to consolidate javadocs later, and add thrown exceptions

**Review by bkompis — CHANGES_REQUESTED** [review]
Remove duplicit enum and my authorship

File: api/src/main/java/cz/muni/fi/pa165/mushrooms/enums/MushroomType.java (hunk 1 above)
Comment (Buvko) [reply api/src/main/java/cz/muni/fi/pa165/mushrooms/enums/MushroomType.java]: Are you sure we want to add this dependency?

**Review by bkompis — APPROVED** [review]
OK now, if you don't want to remove my authorship

**Pull request merged**